ddtrace/opentelemetry/_span.py

```diff
@@ -243,8 +251,9 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             if self._record_exception:
                 self.record_exception(exc_val)
             if self._set_status_on_exception:
-                # do not overwrite the status message set by record exception
-                self.set_status(StatusCode.ERROR)
+                # Ensure span is marked as an error span and error attributes are set
+                # This will ensure OpenTelemetry spans are compatible with Error Tracking
+                self._ddspan.set_exc_info(exc_type, exc_val, exc_tb)
```

Comment:
I don't necessarily understand this line, but otherwise the changes in this file look good